### DISCUSSION TITLE
How do I call the element of a struct in an array?

### DISCUSSION THREAD

**Discussion by ndgsghdj** [sst-inc/help#28] [Q&A - iOS] [answered]
iOS Lesson 3
I was trying to see how I could achieve getting the nameLabel in the new segue screen to show the name element of the cell in the tableViewController.
this is my code for the View Controller concerning the segued screen showing the details of the cell
```
import UIKit

var name = String()
var age = Int()

class DetailViewController: UIViewController {
    
    @IBOutlet weak var nameLabel: UILabel!
    @IBOutlet weak var ageLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        nameLabel.text = "\(name)"
        ageLabel.text = "\(age)"
    }
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
    // Get the new view controller using segue.destination.
    // Pass the selected object to the new view controller.
    if segue.identifier == "showDetails" {
    let destinationViewController = segue.destination as! DetailViewController
    // Send data here
    }
    }
```
I thought of trying to call an element in the struct (eg. Classmate.name), however it does not work.
This is the code for the data.swift file:
```
import Foundation
struct Classmate {
    var name: String
    var age: Int
}
var ClassmatesArray = [
    Classmate(name: "Eben Lim", age: 12),
    Classmate(name: "Nikola Winata", age: 12)
]
```

**Comment by granwyntan** [comment, marked as answer]
Hi @ndgsghdj,
`Classmate.name` wouldn't work because you have declared the Classmate struct type that contains two properties: name as well as age. You have then created multiple instances of the struct in the array, ClassmatesArray, so in order to access and read the values of the properties, you would then need to specify an index within the array.
An example would be:
```swift
var index = 0
...
override func viewDidLoad() {
    super.viewDidLoad()
    nameLabel.text = "\(ClassmatesArray[index].name)"
    ageLabel.text = "\(ClassmatesArray[index].age)"
}
```
Hope this helps!

**Reply by ndgsghdj** [reply]
When doing that, I get an error saying that I "Cannot convert value of type '(Any) -> Int' to expected argument type 'Int'" for both nameLabel.text and ageLabel.text
```
import UIKit
var index = 0
class DetailViewController: UIViewController {
    
    @IBOutlet weak var nameLabel: UILabel!
    @IBOutlet weak var ageLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        nameLabel.text = "\(ClassmatesArray[index].name)"
        ageLabel.text = "\(ClassmatesArray[index].age)"
    }
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
    // Get the new view controller using segue.destination.
    // Pass the selected object to the new view controller.
    if segue.identifier == "showDetails" {
    let destinationViewController = segue.destination as! DetailViewController
    // Send data here
    }
    }
```
<img width="478" alt="Screenshot 2021-08-09 at 2 47 00 PM" src="https://user-images.githubusercontent.com/75611248/128668965-90c54b80-b852-4393-a11a-ae45ce884083.png">

**Reply by granwyntan** [reply]
What line would that be on, could u take a full screenshot or send the zipped project, thanks!

**Reply by ndgsghdj** [reply]
The error is on line 17-18
<img width="897" alt="Screenshot 2021-08-09 at 4 22 37 PM" src="https://user-images.githubusercontent.com/75611248/128677856-69d08da2-9946-4c9b-92c6-777faf8e8c9b.png">

**Reply by granwyntan** [reply]
The prepare(for segue:, sender:) function should be changed in the previous view controller, and could you try declaring index within DetailViewController
Hope it helps!

**Reply by ndgsghdj** [reply]
I have tried declaring the index as a var at the top of the code, but the error still persists
<img width="896" alt="Screenshot 2021-08-09 at 9 12 47 PM" src="https://user-images.githubusercontent.com/75611248/128711861-40526e27-e8dd-4888-8e58-889bee5a672c.png">
Here is my source code:
[Contacts List 2.zip](https://github.com/sst-inc/help/files/6954778/Contacts.List.2.zip)

**Reply by granwyntan** [reply]
Have u tried declaring index within the class DetailViewController, i.e. at line 11?

**Reply by ndgsghdj** [reply]
Oh, it works, thanks

**Reply by granwyntan** [reply]
Great👍
